ddtrace/appsec/_asm_request_context.py

```diff
@@ -156,7 +156,7 @@ def finalise(self):
                 if len(callbacks) > 0:
                     for function in callbacks:
                         function(env)
-                    self.execution_context.end()
+                self.execution_context.end()
```

Comment:
@emmettbutler just an note that this is commit introduces a change a behavior from previous code, since `self.execution_context.end()` was being called inside the `if callbacks:` (now `if len(callbacks) > 0:`) clause

Reply:
(referencing https://github.com/DataDog/dd-trace-py/pull/8646/commits/e1b1ae871483789eb2d07ff9adae769c0c2bda4b specifically)

Reply:
Good catch, thanks